### PR TITLE
Changes to allow Travis CI and Coveralls integration.

### PR DESCRIPTION
**Pull request by rj-cooper**
Fixes the default .travis.yml so that this can be used with dotnet core. This allows automated CI builds using https://travis-ci.org and test coverage using https://coveralls.io

See [here ](https://travis-ci.org/frozenskys/generator-cake) and [here ](https://coveralls.io/github/frozenskys/generator-cake) for examples using my fork.

Also includes the fix for the cake-build resources in the unit tests from [PR#2](https://github.com/agc93/generator-cake/pull/2)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,7 +53,6 @@
   },
   "repository": "agc93/generator-cake",
   "scripts": {
-    "prepublish": "gulp prepublish",
```
Comment (agc93) [inline package.json:55]: Why remove this?

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,11 @@
+sudo: required
+dist: trusty
 language: node_js
 node_js:
   - v6
   - v5
-  - v4
-  - '0.12'
-  - '0.10'
+before_install:
```
Comment (agc93) [inline .travis.yml:7]: Does using the `csharp`/`dotnet` image (as per [these docs](https://docs.microsoft.com/en-us/dotnet/articles/core/tools/using-ci-with-cli) ) not work?

Comment (rj-cooper) [reply .travis.yml:7]: Changing the language as per the [docs](https://docs.microsoft.com/en-us/dotnet/articles/core/tools/using-ci-with-cli) changes the script that is run and the environment variables that are setup. The C# image is designed to build .net core projects and would need some work to build node. As generator-cake is an npm package I decided to go with the node_js language and use the install instructions from https://www.microsoft.com/net/core#ubuntu 

Do you think there is an advantage in using the `csharp / dotnet` image? If so I'll put some more effort into getting it to run.

**Comment by rj-cooper** [comment]
I'll fix the `gulp prepublish` error with another commit. Let me know if you think it's worth putting more effort into the `csharp / dotnet` image for travis.

File: package.json (hunk 1 above)
Comment (rj-cooper) [reply package.json:55]: Hi Alistair, This was an oversight on my part and should not have been removed - I work both remotely and in our office, however when in the office we have an SSL intercepting proxy, which as yet I have not managed to get working with gulp nsp (called by gulp pre-publish) or with npm install from a git repo. Usually I add this back in before I commit, but missed it this time.

**Comment by rj-cooper** [comment]
So the below travis file works, but would need a little more work to build build against multiple node versions.

```
sudo: required
dist: trusty
language: csharp
mono: none
dotnet: 1.0.0-preview2-003131
env: DOTNETCORE=1
install:
  - nvm install node
  - npm install
script:
  - gulp
```

File: package.json (hunk 1 above)
Comment (agc93) [reply package.json:55]: Ah all good, I know the pain of working behind the corporate SSL proxy!

File: .travis.yml (hunk 2 above)
Comment (agc93) [inline .travis.yml]: No, that's fine. I haven't actually used Travis before so am running purely off their documentation for this. I'll take your wisdom here.

**Comment by agc93** [comment]
I haven't used Travis before but it seems to me like building against multiple Node versions is probably the better call here (Node certainly seems to change more than .NET). Any objections @frozenskys ?

I'll get started setting up Travis and Coveralls for this repo as soon as this is merged in.

**Comment by rj-cooper** [comment]
Having looked further into it the `csharp / dotnet` image would appear to run the same commands to install dotnet as my original travis file, so that's probably the best one to go with. That file (the same one as in this PR) builds for node v5 and v6 (I couldn't get `eslint-config-xo-space` to work on any version below 5)  and dotnet core 1.0.1. Adding a node version is just a matter of adding that version to the node_js section, and they will build in parallel. 

```
sudo: required
dist: trusty
language: node_js
node_js:
  - v6
  - v5
before_install:
  - sudo sh -c 'echo "deb [arch=amd64] https://apt-mo.trafficmanager.net/repos/dotnet-release/ trusty main" > /etc/apt/sources.list.d/dotnetdev.list'
  - sudo apt-key adv --keyserver apt-mo.trafficmanager.net --recv-keys 417A0893
  - sudo apt-get -qq update
  - sudo apt-get install dotnet-dev-1.0.0-preview2-003131 -y
```

**Comment by rj-cooper** [comment]
There is also a nice workflow for travis releases where you can add a section like

```
deploy:
  provider: npm
  email: me@mynpmemail.com
  api_key:
    secure: my_encrypted_npmjs_key
  on:
    tags: true
    repo: username/repo-name
```

to your travis.yml file and it will build and publish to npmjs when you do a

```
git tag mytag && git push --tags
```

or 

```
npm version minor && git push --tags
```

see https://docs.travis-ci.com/user/deployment/npm/ for more details.

**Comment by agc93** [comment]
Oh okay cool. I have been using my personal VSTS to do continuous build and deploy (I'm a .NET guy so I was more familiar with it), but I might look into that as well, because it does look like a good workflow.

Alright, if you're happy with the `travis.yml` as it stands, I'm happy to merge this in, and I'll set up Travis and Coveralls on the repo with the new config.

As always, thanks for the awesome contribution 😄

**Comment by agc93** [comment]
And I've updated the [README](https://github.com/agc93/generator-cake) with build and coverage status.

Thanks again @frozenskys !

**Comment by rj-cooper** [comment]
Just glad to be able to contribute a little back to a great project!

I'm also a .NET guy (hence the interest in cake and generator-cake) but part of my job is to look for cool stuff that will help the team perform better and automate more.

If you get five minutes please can you push the latest version to npmjs, as we can't pull node modules directly from Github due to our proxy, and my cool demo of yeoman needs the compose functionality in generator-cake for it to work.

**Comment by agc93** [comment]
Yep, will do tonight (i.e. within the next 6 hours), just can't push from work thanks to our corporate proxy being difficult.

**Comment by rj-cooper** [comment]
Yeah - I feel your pain - you could make a fortune if you could fix the "Corporate Proxy" issue...

**Comment by agc93** [comment]
Meeting room wifi doesn't have a proxy, so 0.2.3 is now live! :shipit: 

Let me know if you see any issues.

**Comment by rj-cooper** [comment]
Works like a dream :) Thanks!

**Pull request merged**